content/courses/native-onchain-development/serialize-instruction-data-frontend.md

```diff
@@ -359,9 +372,14 @@ export class Movie {
   ])
 
   serialize(): Buffer {
-    const buffer = Buffer.alloc(1000)
-    this.borshInstructionSchema.encode({ ...this, variant: 0 }, buffer)
-    return buffer.slice(0, this.borshInstructionSchema.getSpan(buffer))
+    try {
+      const buffer = Buffer.alloc(1000);
```

Comment:
We should explain why we're picking 1000 here.

Reply:
done

Reply:
We should be able to know the size of a movie review based on INIT_SPACE plus ANCHOR_DISCRIMINATOR.

Reply:
> We should be able to know the size of a movie review based on INIT_SPACE plus ANCHOR_DISCRIMINATOR.

implemented ✅